### PR TITLE
render: Update wgpu to 23

### PR DESCRIPTION
**Pull request by torokati44**
This lets us drop a whole bunch of (duplicated) dependencies, just look at Cargo.lock!
Currently using egui master.

### CODE REVIEW

**Comment by Dinnerbone** [comment]
I'm hoping to see a little improvement to webgl performance with this!

**Comment by torokati44** [comment]
@adrian17 I saw you made some recommendations for this on Discord - unfortunately I am weeks behind reading a lot of the channels there, and don't really want to skip over anything important - so if you could please copy them over here, or even amend the PR if you have the time, I would be grateful!

One suggestion was about not naming the shader stages, right? None at all, or just some?

@Dinnerbone I assume you mean this, right: https://github.com/gfx-rs/wgpu/pull/6427 (Just for future reference...)

File: desktop/src/util.rs [hunk 1]
```diff
@@ -254,27 +254,10 @@ pub fn plot_stats_in_tracy(instance: &wgpu::Instance) {
...
-    if let Some(stats) = backend {
-        tracy.plot(BIND_GROUPS, stats.bind_groups.num_allocated as f64);
-        tracy.plot(BUFFERS, stats.buffers.num_allocated as f64);
-        tracy.plot(TEXTURES, stats.textures.num_allocated as f64);
-        tracy.plot(TEXTURE_VIEWS, stats.texture_views.num_allocated as f64);
-    }
+    tracy.plot(BIND_GROUPS, report.hub.bind_groups.num_allocated as f64);
+    tracy.plot(BUFFERS, report.hub.buffers.num_allocated as f64);
+    tracy.plot(TEXTURES, report.hub.textures.num_allocated as f64);
+    tracy.plot(TEXTURE_VIEWS, report.hub.texture_views.num_allocated as f64);
```
Comment (torokati44) [inline desktop/src/util.rs:260]: About this, see: 

https://github.com/gfx-rs/wgpu/pull/6100/files#diff-a7313a05e0631391e737b41d80d6639bc143a5210b8354d7faed6342a76093c7R11-R12

https://github.com/gfx-rs/wgpu/pull/6100/files#diff-a7313a05e0631391e737b41d80d6639bc143a5210b8354d7faed6342a76093c7R24-R28

Comment (torokati44) [reply desktop/src/util.rs:260]: Bah, GH linking...
Basically:
image

**Review by Dinnerbone — APPROVED** [review]

**Pull request merged**